plugins/terminal.koplugin/profile

```diff
@@ -7,6 +7,11 @@
 
 # shellcheck source=/dev/null
 
+if [ -x "${SHELL}" ] && [ "${SHELL}" != "$0" ]; then
```

Comment:
Also looking at the code a bit more checking if it's executable might be a bit silly, or in any perhaps more fitting in Lua?

For example something like:

```
diff --git a/plugins/terminal.koplugin/main.lua b/plugins/terminal.koplugin/main.lua
index a94d9bc42..b8cc52d63 100644
--- a/plugins/terminal.koplugin/main.lua
+++ b/plugins/terminal.koplugin/main.lua
@@ -111,7 +111,7 @@ function Terminal:spawnShell(cols, rows)
         return true
     end
 
-    local shell = G_reader_settings:readSetting("terminal_shell", "sh")
+    local shell = G_reader_settings:readSetting("terminal_shell", os.getenv("SHELL") or "sh")
 
     local ptmx_name = "/dev/ptmx"
     self.ptmx = C.open(ptmx_name, bit.bor(C.O_RDWR, C.O_NONBLOCK, C.O_CLOEXEC))
```

I don't really mind one way or the other, but it optimizes away needlessly starting `sh` and it keeps it all in one place.

Of course if you did want to keep the executable check this is a lot more legible than the lfs equivalent…

Reply:
PS Empty string is truthy I believe, bad pseudocode. ;-)

Reply:
I'm not immediately sure what you're referring to. I just meant that `os.getenv("SHELL")` might be an empty string.